purchase_request/models/purchase_request.py

```diff
@@ -67,6 +68,14 @@ def _track_subtype(self, init_values):
                 return 'purchase_request.mt_request_done'
         return super(PurchaseRequest, self)._track_subtype(init_values)
 
+    @api.multi
+    @api.constrains('picking_type_id')
+    def _check_picking_type_id(self):
+        for rec in self:
+            if rec.picking_type_id.code != 'incoming':
+                raise UserError(_(
```

Comment:
it's just more significant, on a constraint a ValidationError is more appropriate than a UserError